contracts/RFQ.sol

```diff
@@ -123,15 +159,33 @@ contract RFQ is IRFQ, StrategyBase, ReentrancyGuard, SignatureValidator, BaseLib
     }
 
     // settle
-    function _settle(RFQLibEIP712.Order memory _order, GroupedVars memory _vars) internal returns (uint256) {
+    function _settle(
+        RFQLibEIP712.Order memory _order,
+        SpenderLibEIP712.SpendWithPermit memory _makerAssetPermit,
+        SpenderLibEIP712.SpendWithPermit memory _takerAssetPermit,
+        GroupedVars memory _vars,
+        bytes memory _makerAssetPermitSig,
+        bytes memory _takerAssetPermitSig
+    ) internal returns (uint256) {
         // Transfer taker asset to maker
         if (address(weth) == _order.takerAssetAddr) {
             // Deposit to WETH if taker asset is ETH
             require(msg.value == _order.takerAssetAmount, "RFQ: insufficient ETH");
             weth.deposit{ value: msg.value }();
             weth.transfer(_order.makerAddr, _order.takerAssetAmount);
         } else {
-            spender.spendFromUserTo(_order.takerAddr, _order.takerAssetAddr, _order.makerAddr, _order.takerAssetAmount);
+            require(
+                // Confirm that 'takerAddr' sends 'takerAssetAmount' amount of 'takerAssetAddr' to 'address(this)'
+                _order.takerAddr == _takerAssetPermit.user &&
+                    _order.takerAssetAddr == _takerAssetPermit.tokenAddr &&
+                    address(this) == _takerAssetPermit.recipient &&
+                    _order.takerAssetAmount == _takerAssetPermit.amount,
+                "RFQ: taker spender information is incorrect"
```

Comment:
You fill in the permit data with data from `_order` so there's no need to check them again here.